### PR TITLE
Adding feature to retry terraform apply on any failure

### LINKED ISSUE #232: Track upstream `aws_s3_bucket_logging` bug report and remove workaround
### Persona

Maintainer

### Description

Need to monitor this issue for provider updates/fixes
https://github.com/hashicorp/terraform-provider-aws/issues/31139

### Use Case

Will remove (future) barnacles when issue is resolved

### Impact

Barnacles in test pipeline

### Completion

1. Provider is fixed
2. We update version in our codebase
3. Remove retry logic in terratest
4. No more intermittent errors in pipeline due to `aws_s3_bucket_logging` resource

### Additional Context

related: https://github.com/defenseunicorns/delivery-aws-iac/issues/207

Barnacle verbiage: unnecessary or outdated code/features

### PR DESCRIPTION
**Pull request by zack-is-cool**
closes #232

Pivoting to catch all errors (specifically transient), retry 5 times on any failure. Doesn't introduce and risk.

### CODE REVIEW

File: test/e2e/examples_complete_insecure_test.go [hunk 1]
```diff
@@ -21,8 +21,7 @@ func TestExamplesCompleteInsecure(t *testing.T) {
 			"fixtures.insecure.tfvars",
 		},
 		RetryableTerraformErrors: map[string]string{
-			".*empty output.*": "bug in aws_s3_bucket_logging, intermittent error",
-			".*timeout while waiting for state to become 'ACTIVE'.*": "Sometimes the EKS cluster takes a long time to create",
+			".*": "Failed to apply Terraform configuration due to an error.",
```
Comment (RothAndrew) [inline test/e2e/examples_complete_insecure_test.go:24]: Why is this change needed? Are there a lot more errors that are happening?

Comment (zack-is-cool) [reply test/e2e/examples_complete_insecure_test.go:24]: Instead of chasing intermittent errors with the terraform provider/AWS api this would just be a catchall. Anywhere that there would be an actual issue - it would still fail. 

We could still see the retries pipeline, but it wouldn't be "oh this failed due to that thing that sometimes happens, let me run the pipeline again and come back in 70 minutes."

Comment (RothAndrew) [reply test/e2e/examples_complete_insecure_test.go:24]: > "oh this failed due to that thing that sometimes happens"

Couldn't we add the thing to the list of retryable errors?

Switching to a catchall would mean errors that are not retryable are retried too.

Comment (RothAndrew) [reply test/e2e/examples_complete_insecure_test.go:24]: @ntwkninja curious to hear your take. I don't love this solution but I'm also not so completely against it that I would block it from getting merged.

Comment (zack-is-cool) [reply test/e2e/examples_complete_insecure_test.go:24]: If you look through the past jobs for this PR you'll see a few other intermittent errors
1. https://github.com/defenseunicorns/delivery-aws-iac/actions/runs/5590108266/job/15140531686#step:3:17196
2. https://github.com/defenseunicorns/delivery-aws-iac/actions/runs/5593003123/job/15149414941#step:3:26942
3. https://github.com/defenseunicorns/delivery-aws-iac/actions/runs/5593003123/job/15149414941#step:3:26942

Comment (RothAndrew) [reply test/e2e/examples_complete_insecure_test.go:24]: We could add new lines to the array that retry when "Please retry" appears or when "empty result" appears. I agree both of those are retryable errors.

Comment (zack-is-cool) [reply test/e2e/examples_complete_insecure_test.go:24]: > > "oh this failed due to that thing that sometimes happens"
> 
> Couldn't we add the thing to the list of retryable errors?
> 
> Switching to a catchall would mean errors that are not retryable are retried too.

Right, and imo worst case with that is it just does another terraform apply and it fail 5 times. I feel like that's a good trade off vs having to spend hours debugging and adding different error messages to our retry list.

Comment (RothAndrew) [reply test/e2e/examples_complete_insecure_test.go:24]: Okay, yeah I think you are right the world won't come crashing down if we retry in instances where we don't need to. I don't see any harm in merging this and seeing how things go.

**Review by RothAndrew — APPROVED** [review]
